flag/flags.go

```diff
@@ -0,0 +1,12 @@
+package flag
+
+// Flags provides the ability to create and communicate feature flags.
+type Flags uint64
+
+// Flags is a bitset
+const (
+	Debug Flags = 1 << iota
+	SamplingSet
```

Comment:
note that everything beyond Debug isn't relevant unless you are doing binary, fixed-width propagation. Ex. Sampled seems to be stored in Context.sampled and IsRoot isn't needed since ParentSpanID is a ref